pkg/auth/dependency/newinteraction/node_select_identity_end.go

```diff
@@ -0,0 +1,38 @@
+package newinteraction
+
+import "github.com/authgear/authgear-server/pkg/auth/dependency/identity"
+
+type EdgeSelectIdentityEnd struct {
+	Identity *identity.Info
+}
+
+func (e *EdgeSelectIdentityEnd) Instantiate(ctx *Context, graph *Graph, rawInput interface{}) (Node, error) {
+	input, ok := rawInput.(InputSelectIdentityBegin)
+	if !ok {
+		return nil, ErrIncompatibleInput
+	}
+
+	return &NodeSelectIdentityBegin{
+		UseAnonymousUser: input.GetUseAnonymousUser(),
+	}, nil
+}
+
+type NodeSelectIdentityEnd struct {
+	Identity *identity.Info `json:"identity"`
+}
+
+func (n *NodeSelectIdentityEnd) Apply(ctx *Context, graph *Graph) error {
+	panic("implement me")
+}
+
+func (n *NodeSelectIdentityEnd) DeriveEdges(ctx *Context, graph *Graph) ([]Edge, error) {
+	edge, err := graph.Intent.(interface {
+		AfterSelectIdentity(node *NodeSelectIdentityEnd) (Edge, error)
+	}).AfterSelectIdentity(n)
+
+	return []Edge{edge}, err
```

Comment:
I think it is better to add `DeriveEdges(ctx *Context, graph *Graph, node Node) ([]Edge, error)` to interface `Intent`. So here we just write

```golang
func (n *NodeSelectIdentityEnd) DerivesEdges(ctx *Context, graph *Graph) ([]Edge, error) {
  return graph.Intent.DerivesEdges(ctx, graph, n)
}
```